servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ClientParentChannelInitializer.java

```diff
@@ -72,7 +84,14 @@ private H2PushStreamHandler() {
 
         @Override
         public void channelRegistered(final ChannelHandlerContext ctx) {
-            ctx.close(); // push streams are not supported
+            // See SETTINGS_ENABLE_PUSH in https://tools.ietf.org/html/rfc7540#section-6.5.2
+            ctx.writeAndFlush(new DefaultHttp2GoAwayFrame(PROTOCOL_ERROR))
+                    .addListener((ChannelFutureListener) future -> {
+                if (!future.isSuccess()) {
+                    LOGGER.debug("Failed to send a GO_AWAY frame for received PUSH_PROMISE frame", future.cause());
+                }
+                ctx.close(); // push streams are not supported
```

Comment:
Netty should be closing the channel and log appropriately:

https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java#L905-L929

I do not think we need to add a listener here